File: snippets/csharp/recommend_points.cs
using Qdrant.Client;
using Qdrant.Client.Grpc;
using static Qdrant.Client.Grpc.Conditions;

var client = new QdrantClient("localhost", 6334);

await client.RecommendAsync(
    "{collection_name}",
    positive: new ulong[] { 100, 231 },
    negative: new ulong[] { 718 },
    filter: MatchKeyword("city", "London"),
    limit: 3
);

File: snippets/go/recommend_points.go
package client

import (
	"context"

	"github.com/qdrant/go-client/qdrant"
)

func recommend() {
	client, err := qdrant.NewClient(&qdrant.Config{
		Host: "localhost",
		Port: 6334,
	})
	if err != nil {
		panic(err)
	}

	client.Query(context.Background(), &qdrant.QueryPoints{
		CollectionName: "{collection_name}",
		Query: qdrant.NewQueryRecommend(&qdrant.RecommendInput{
			Positive: []*qdrant.VectorInput{
				qdrant.NewVectorInputID(qdrant.NewIDNum(100)),
				qdrant.NewVectorInputID(qdrant.NewIDNum(231)),
			},
			Negative: []*qdrant.VectorInput{
				qdrant.NewVectorInputID(qdrant.NewIDNum(718)),
				qdrant.NewVectorInput(0.2, 0.3, 0.4, 0.5),
			},
			Strategy: qdrant.RecommendStrategy_AverageVector.Enum(),
		}),
		Filter: &qdrant.Filter{
			Must: []*qdrant.Condition{
				qdrant.NewMatch("city", "London"),
			},
		},
	})
}
